### ISSUE TITLE
gents.TypeScriptGeneratorTests fail

### ISSUE THREAD

**Issue by rehmsen** [angular/clutz#342]
I checked out clean master and ran `gradle test`, and got 4 failed tests (full output attached). Throwing this into meld, The issue is with newlines.
[test_output.txt](https://github.com/angular/clutz/files/484396/test_output.txt)

Expected:
```javascript
constructor(a: number) {[ this.a = a; }
```
Got:
```javascript
constructor(a: number) {[
   this.a = a;
}
```

Rest similar. What I do not understand is what the square brackets are doing in the output. Neither the input nor the golden file have these.

**Comment by rehmsen** [comment]
The [ seem to be a glitch in the output - when I change the golden file to break lines even on short methods, the tests pass. 

I am not sure what the expectation is here- is Gents expected to keep short methods in one line or not? That would inform if we need to just fix the golden files, or if we need to fix the code.

**Comment by mprobst** [comment]
I think this is clang-format being newer or older on the build machine than expected. The `[` is just the indicator from the diff of what fails?

**Comment by rehmsen** [comment]
Okay - maybe because I have a google version of clang format. Seems suboptimal that the unit test would depend on the clang format you have installed on your system... maybe the action here is to not use local clang-format but isolate that?

**Comment by mprobst** [comment]
I think the actual problem (tm) is that your local environment is not up to date. Could you try running `rm -r node_modules && npm install clang-format typescript@beta`? That fixes the problem for me.

**Comment by rehmsen** [comment]
No, the problem is that we run the system-wide clang-format. That is a bad idea, unit tests should isolate from our system environment. And apparently we are trying to do the right thing: We are installing clang-format locally, so it ends up in node_modules. But then we run the one on the path instead of the one we just installed.

I was able to hard code the one in node_modules in TypeScriptGenerator.java and then the test pass. So now I need to figure out how to make it select the local one (without hardcoding linux_x64 etc.).
